frontend/src/components/CognitoLoginAndReg.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import Button from 'react-bootstrap/Button';
+import { useLocation } from 'react-router-dom'
+
+function useQuery() {
+  return new URLSearchParams(useLocation().search);
+}
+
+export default () => {
+  const [, setTokenObj ] = useState({});
+  const [, setUserObj ] = useState({});
+
+  let query = useQuery();
+  let code = query.get("code");
+
+  function fetchTokens() {
+    const myHeaders = new Headers();
+    myHeaders.append("Content-Type", "application/x-www-form-urlencoded");
+
+    const urlencoded = new URLSearchParams();
+    urlencoded.append("grant_type", "authorization_code");
+    urlencoded.append("client_id", "133b172mr6hvdr48eg0mb2uirh");
+    urlencoded.append("code", `${code}`);
+    urlencoded.append("redirect_uri", "http://localhost:5004/login");
+
+    const requestOptions: any = {
+      method: 'POST',
+      headers: myHeaders,
+      body: urlencoded,
+      redirect: 'follow'
+    };
+
+    return fetch("https://infomed1.auth.us-east-2.amazoncognito.com/oauth2/token", requestOptions)
+      .then((response: any) => response.json())
+  }
+
+  function fetchUserInfo (access_token: any) {
+    var myHeaders = new Headers();
+    myHeaders.append("Authorization", `Bearer ${access_token}`);
+
+    var requestOptions: any = {
+      method: 'GET',
+      headers: myHeaders,
+      redirect: 'follow'
+    };
+
+    return fetch("https://infomed1.auth.us-east-2.amazoncognito.com/oauth2/userInfo", requestOptions)
+      .then((response: any) => response.json())
+  }
+
+  useEffect (() => {
+
+    fetchTokens()
+      .then((tokenObject) => {
+        setTokenObj(tokenObject);
+        return fetchUserInfo(tokenObject.access_token);
+      })
+      .then((userInfo) => {
+        setUserObj(userInfo);
+        console.log(userInfo);
+      })
+      .catch((error) => console.log(error));
+  }, [code]);
+
+  return (
+    <div className="text-center">
+      <Button 
+        className="mt-3" 
+        type="submit" 
+      >
+        <a href="https://infomed1.auth.us-east-2.amazoncognito.com/login?client_id=133b172mr6hvdr48eg0mb2uirh&response_type=code&scope=email+openid&redirect_uri=http://localhost:5004/login">
+          Login
+        </a>
```

Comment:
A link inside a button that submits a form? Why? (By the way, it would submit a form which is not in this component. It may have unpredictable consequences, depending on where this component is placed.)

Reply:
This isn't the actual page where the login redirect would happen from, this buttons were placed here simply for testing purposes. The actual login buttons will be elsewhere, maybe on the landing page or somewhere else.

Reply:
The buttons are not for form submissions, they are just for redirecting to the Cognito URL.